sympy/ntheory/modular.py

```diff
@@ -250,3 +250,45 @@ def combine(c1, c2):
         if symmetric:
             return symmetric_residue(n, m), m
         return n, m
+
+def mod_inverse(a, m):
+    """
+    Given two number a  and m.This function computes the number
+    c such that   (a*c)%m==1.
+    Meaning that if a is multiplied by c then the remainder obtained
+    on division with m is 1.  The value of m need not be a prime.
+    This function  returns  the value of c  if some c exists.
+    Otherwies it raises an Exception stating that modular inverse does not
+    exist.
+
+    Examples
+    ========
+    From https://en.wikipedia.org/wiki/Modular_multiplicative_inverse
+
+    Suppose we wish to find modular multiplicative inverse x of 3 modulo 11.
+    This is the same as finding x such that
+    3*x = 1 (mod 11)
+    we find one value of x that satisfies this congruence is 4
+    Because 3*4=12   and 12 = 1 mod(11)
+    So          modular_inverse(3,11)=4
+
+    similarly   modular_inverse(5,11)=9
+                modular_inverse(21124921,521512)=7713
```

Comment:
Use Python interpreter style doctests.
Check other modules for similar docstrings